src/NuGet.Clients/NuGet.CommandLine/Commands/InstallCommand.cs

```diff
@@ -238,28 +258,33 @@ private DependencyBehavior GetDependencyBehavior()
             return dependencyBehavior;
         }
 
-        private async Task InstallPackage(
+        private async Task InstallPackageAsync(
             string packageId,
             NuGetVersion version,
             string installPath)
         {
             if (version == null)
             {
-                NoCache = true;
+                // Avoid searching for the highest version in the global packages folder,
+                // it needs to come from the feeds instead. Once found it may come from
+                // the global packages folder unless NoCache is true.
+                ExcludeCacheAsSource = true;
```

Comment:
this seems a bit arbitrary. how will people realize what is the resolution logic?